src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * @flow
+ * @file Info component used by AppActivity component
+ */
+
+import * as React from 'react';
+import { FormattedMessage } from 'react-intl';
+import classNames from 'classnames';
+import getProp from 'lodash/get';
+import noop from 'lodash/noop';
+
+import InlineDelete from '../inline-delete';
+import { ReadableTime } from '../../../../components/time';
+import Tooltip from '../../../../components/tooltip';
+import { Link } from '../../../../components/link';
+import messages from '../../../common/messages';
+
+import './AppActivity.scss';
+
+const ONE_HOUR_MS = 3600000; // 60 * 60 * 1000
+const parser = new DOMParser();
+
+type Props = {
+    activity_template: Object,
+    app: Object,
+    created_at: string,
+    created_by: User,
+    currentUser: User,
+    error?: ActionItemError,
+    id: string,
+    isPending: boolean,
+    onDelete: Function,
+    permissions?: BoxItemPermission,
+    rendered_text: string,
+};
+
+function parseActivity(htmlString: string): Array<HTMLElement> {
+    const doc = parser.parseFromString(htmlString, 'text/html');
+    return Array.from(doc.body.childNodes);
+}
+
+function mapActivityNodes(node: HTMLElement): React.Node {
+    const { dataset = {}, href = '#', tagName, textContent } = node;
+    switch (tagName) {
+        case 'A':
+            return (
+                <Link
+                    href={href}
+                    data-resin-target={dataset.resinTarget}
+                    data-resin-action={dataset.resinAction}
+                    rel="roreferrer noopener"
+                    className="bcs-app-activity-link"
+                >
+                    {textContent}
+                </Link>
+            );
+        default:
+            return textContent;
+    }
+}
+
+const AppActivity = ({
+    activity_template,
+    app,
+    currentUser,
+    created_at,
+    created_by,
+    error,
+    id,
+    isPending,
+    rendered_text,
+    permissions = {},
+    onDelete = noop,
+}: Props): React.Node => {
+    const createdAtTimestamp = new Date(created_at).getTime();
+    const activityNodes = parseActivity(rendered_text);
+    const canDelete = getProp(permissions, 'can_delete', false);
+    const { name, icon_url } = app;
+    const { id: templateId } = activity_template;
+
+    return (
+        <div
+            className="bcs-app-activity-container"
+            data-resin-target="loaded"
+            data-resin-feature={`app_activity_${templateId}_card`}
+        >
+            <div
+                className={classNames('bcs-app-activity', {
+                    'bcs-is-pending': isPending || error,
```

Comment:
What does `isPending` mean app activity? Is it as a pending state when it is being deleted? Is the error state the same as pending state? Would it make more semantic sense if we had a class to distinguish an error state?

Reply:
The item will be pending when being deleted. I was following what comments do, in regard to errors. I think what we can do here is take up a task to create UI to better distinguish errors for a particular item in the ActivityFeed.